### PR TITLE
update heartbeat nodes

### PR DESCRIPTION
**Pull request by HagarMeir**
Change-Id: I33a8deb28020376d35de48a0c410de6541334d6a
Signed-off-by: Hagar Meir <hagar.meir@ibm.com>

### CODE REVIEW

File: orderer/consensus/smartbft/chain.go [hunk 1]
```diff
@@ -196,22 +196,32 @@ func NewChain(
...
+				break
+			}
+		}
+		if receiver == false {
+			heartbeatSenders = append(heartbeatSenders, node)
 		}
 	}
-	c.heartbeatMonitor = NewHeartbeatMonitor(c.consensus.Comm.(MessageSender), heartbeatTicker.C, logger, heartbeatTimeout, heartbeatCount, role, senders, receivers)
+
+	c.heartbeatMonitor = NewHeartbeatMonitor(c.consensus.Comm.(MessageSender), heartbeatTicker.C, logger, heartbeatTimeout, heartbeatCount, myRole, heartbeatSenders, heartbeatReceivers)
```
Comment (yacovm) [inline orderer/consensus/smartbft/chain.go:222]: when we change a committee upon a block commit, we need to re-create this monitor.

File: orderer/consensus/smartbft/chain.go [hunk 2]
```diff
@@ -196,22 +196,32 @@ func NewChain(
...
-			senders = append(senders, uint64(node.ID))
-		} else {
-			receivers = append(receivers, uint64(node.ID))
+		if selfID == uint64(node.ID) {
+			myRole = HeartbeatReceiver
+		}
+		heartbeatReceivers = append(heartbeatReceivers, uint64(node.ID))
+	}
+
+	heartbeatSenders := make([]uint64, 0)
```
Comment (yacovm) [inline orderer/consensus/smartbft/chain.go]: please extract this block to a function that returns the senders and the receivers (and your role)

File: orderer/consensus/smartbft/chain.go [hunk 3]
```diff
@@ -196,22 +196,32 @@ func NewChain(
...
+	for _, node := range rtc.Nodes {
+		receiver := false
+		for _, committeeNode := range currentCommittee {
+			if node == uint64(committeeNode.ID) {
+				receiver = true
+				break
+			}
+		}
+		if receiver == false {
```
Comment (yacovm) [inline orderer/consensus/smartbft/chain.go]: `if receiver == false` 

More succinct to **continue** of this is a receiver

File: orderer/consensus/smartbft/chain.go (hunk 1 above)
Comment (HagarMeir) [reply orderer/consensus/smartbft/chain.go:222]: true
what should I do about this now?

File: orderer/consensus/smartbft/chain.go (hunk 3 above)
Comment (HagarMeir) [reply orderer/consensus/smartbft/chain.go]: why?
what do you mean exactly?

File: orderer/consensus/smartbft/chain.go (hunk 1 above)
Comment (yacovm) [reply orderer/consensus/smartbft/chain.go:222]: I suggest you hold a reference to the monitor and upon every block commit, when we compute the new committee, you will make the required adjustments according to whether there was a committee change, or not.

**Pull request closed without merging**